### DISCUSSION TITLE
Procedure to make the HR working on zwift

### DISCUSSION THREAD

**Discussion by benriou** [cagnulein/qdomyos-zwift#211] [Q&A] [answered]
What is the procedure to enable the HR service on zwift ? 
I have the Power Captor, the Cadence Captor, the Resistance Control enabled, but no Heart control. 
I tried the switches '-heart-service', '-no-heart-service' with no changes.

The heart signal is working ok on : BKOOL (FTMS assumption ??) >> detects an HR captor 
The heart signal is not detected on : ZWIFT, FULGAZ.

Client :  Raspberry Pi 0w

Thank you for your time.

**Comment by cagnulein** [comment]
this should be done by -heart-service
Try with this one, what you will see on nrfconnect app. You should see the
heart rate service after the fitness machine service


Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 23 mar 2021 alle ore 14:07 ben75020 ***@***.***>
ha scritto:

**Reply by benriou** [reply]
I can see the difference in the NRF app : 

with -heart-service :  Fitness Machine, Heart Rate
with -no-heart-service : Fitness Machine

However, none of the zwift or fulgaz detects a heart captor. :(

**Comment by cagnulein** [comment]
if nrfconnect it works i guess you have an issue with zwift so.
I just tried it now on QZ on iphone and zwift on PC and it works fine.

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 23 mar 2021 alle ore 14:28 ben75020 ***@***.***>
ha scritto:

**Comment by benriou** [comment]
indeed.
The problem does not happens on zwift Android version, where all four captors (including HR) are detected. 

The problem is only on (my ?) ipad. 
This is an iPad pro 10", I'll have a try with an ipad pro 13" later today

**Reply by cagnulein** [reply]
ok keep me posted, thanks

**Comment by benriou** [comment, marked as answer]
Problem fixed after removing ALL devices paired (but not connected) in BT with the iPad.

I will prepare a troubleshooting guide to the documentation including this matter.

**Comment by cagnulein** [comment]
yeah! great!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 24 mar 2021 alle ore 10:03 ben75020 ***@***.***>
ha scritto:
